velox/core/SimpleFunctionMetadata.h

```diff
@@ -196,15 +208,21 @@ struct TypeAnalysis {
   }
 };
 
-template <typename T>
-struct TypeAnalysis<Generic<T>> {
+template <typename T, bool comparable, bool orderable>
```

Comment:
perhaps, add a static assert that if orderable is true, then comparable is also true

Reply:
I already have it   in the class definition of Generic

template <typename T = AnyType, bool comparable = false, bool orderable = false>
struct Generic {
  Generic() = delete;
  static_assert(!(orderable && !comparable), "Orderable implies comparable.");
};